src/sysc/scc/ordered_semaphore.h

```diff
@@ -155,12 +155,13 @@ class SC_API ordered_semaphore : public sc_core::sc_semaphore_if, public sc_core
                     queue[1].pop_front();
                     return false;
                 }
-            } else {
+            } else if(queue[0].size()) {
                 if(queue[0].front() == sc_core::sc_get_current_process_handle()) {
                     queue[0].pop_front();
                     return false;
                 }
-            }
+            } else
+            	return false;
```

Comment:
residue from old pull-request https://github.com/VP-Vibes/SystemC-Components/pull/4
Not sure where this is from, or what it does. @eyck ?